allennlp/models/model.py

```diff
@@ -211,6 +225,18 @@ def _get_prediction_device(self) -> int:
         else:
             return -1
 
+    def _maybe_warn_for_unseparable_batches(self):
+        """
+        This method warns once if a user implements a model which returns a dictionary with
+        values which we are unable to split back up into elements of the batch. This is controlled
+        by a class attribute ``_warn_for_unseperable_batches`` because it would be extremely verbose
+        otherwise.
+        """
+        if self._warn_for_unseparable_batches:
+            logger.warning("Encountered a key in the model's return dictionary which "
```

Comment:
Maybe have this take the key as an argument, and log that?  Seems like you will _always_ see this for `loss`, right?  You might even just never warn for `loss`.

Reply:
Ah, sorry, `loss` won't trigger this branch.  Still, knowing what key triggered this would be nice.

Reply:
That's actually a good idea - we should warn per key. We want to warn for loss, because it's definitely something that you might want to return, which you could get around (if you were warned about it) by adding a `batch_loss` key to your model output.